symfit/core/minimizers.py

```diff
@@ -405,12 +405,15 @@ def scipy_constraints(self, constraints):
                     raise AssertionError('The constraint should accept the same'
                                          ' parameters as used for the fit.')
                 constraint_type = constraint.constraint_type
-                constraint = MinimizeModel(constraint, data={})
+                constraint = MinimizeModel(constraint, data=self.objective.data)
             elif isinstance(constraint, sympy.Rel):
-                from .fit import Constraint, CallableNumericalModel
-                constraint = Constraint(constraint, params=self.parameters)
+                from .fit import prepare_constraint
+                constraint = prepare_constraint(
+                    constraint, self.parameters,
+                    model_type=self.objective.model.__class__
+                )
```

Comment:
I still don't think this constraint init code should be here. If you want to use Minimizers directly, you need to provide the correct arguments. I do agree that the API of Fit, Model, etc should be as flexible and user friendly as possible.

Reply:
Yes, I kinda agree, but I experienced in my own codebase that this is a very nice feature to have, and not that much work for us.

Reply:
> and not that much work for us.

I disagree there. I think it incurs a rather significant technical debt, but time will tell.

Reply:
We can perhaps review this as a part of #203. I agree that perhaps the minimizer code is becoming bulky, which is definitely not supposed to happen.